skopt/searchcv.py

```diff
@@ -670,14 +677,38 @@ def step(self, X, y, space_id, groups=None, n_jobs=1):
         # feed the point and objective back into optimizer
         local_results = self.cv_results_['mean_test_score'][-len(params):]
 
+        if refit:
+            self._fit_best_model(X, y)
+
         # optimizer minimizes objective, hence provide negative score
-        optimizer.tell(params, [-score for score in local_results])
+        return optimizer.tell(params, [-score for score in local_results])
 
-        # fit the best model if necessary
-        if self.refit:
-            self._fit_best_model(X, y)
 
-    def fit(self, X, y=None, groups=None):
+
+    def total_iterations(self):
```

Comment:
can this be a property please? `@property` should be enough

Reply:
now a property